ln-simln-jamming/src/attacks/utils.rs

```diff
@@ -176,9 +172,371 @@ pub async fn build_reputation<C: Clock + InstantClock, R: ReputationMonitor>(
+    // Took mostly from `setup_three_hop_network_edges` but wit the extra node and channel
+    // specifically for the setup needed here.
+    pub fn setup_four_hop_network_edges() -> (ForwardManagerParams, Vec<NetworkParser>) {
+        let alice = get_random_keypair().1;
+        let bob = get_random_keypair().1;
+        let carol = get_random_keypair().1;
+        let dave = get_random_keypair().1;
+
+        let alice_bob = ShortChannelID::from(1);
+        let alice_carol = ShortChannelID::from(2);
+        let bob_carol = ShortChannelID::from(3);
+        let carol_dave = ShortChannelID::from(4);
+
+        let params = ForwardManagerParams {
+            reputation_params: ReputationParams {
+                revenue_window: Duration::from_secs(60),
+                reputation_multiplier: 60,
+                resolution_period: Duration::from_secs(90),
+                expected_block_speed: None,
+            },
+            general_slot_portion: 30,
+            general_liquidity_portion: 30,
+            congestion_slot_portion: 20,
+            congestion_liquidity_portion: 20,
+        };
+
+        let edges = vec![
+            setup_test_edge(alice_bob, alice, bob),
+            setup_test_edge(alice_carol, alice, carol),
+            setup_test_edge(bob_carol, bob, carol),
+            setup_test_edge(carol_dave, carol, dave),
+        ];
+
+        (params, edges)
+    }
+
+    #[test]
+    fn test_calculated_fee_to_build_reputation() {
+        let (_, target_pubkey) = get_random_keypair();
+        let fwd_params = ForwardManagerParams::default();
+
+        let expiry_delta = 200;
+        let htlc_amounts: Vec<u64> = vec![21_000, 42_000, 100_000];
+        let fee_pct = 0.0001;
+
+        let htlc_routes: Vec<Route> = htlc_amounts
+            .iter()
+            .map(|amount| {
+                build_route_with_target_hop(
+                    target_pubkey,
+                    (*amount as f64 * fee_pct) as u64,
+                    expiry_delta,
+                )
+            })
+            .collect();
+
+        let channel_revenue = 1_000_000;
+        let peer_reputation = 700_000;
+        let fee_buffer = 1_000;
+
+        let (fee_to_pay, total_htlc_risk) = fee_to_build_reputation(
+            &htlc_routes,
+            &fwd_params,
+            target_pubkey,
+            channel_revenue,
+            peer_reputation,
+            fee_buffer,
+        );
+
+        let expected_htlc_risk = {
+            let mut risk = 0;
+            for htlc in htlc_amounts.iter() {
+                risk +=
+                    fwd_params.htlc_opportunity_cost((*htlc as f64 * fee_pct) as u64, expiry_delta);
+            }
+            risk
+        };
+
+        // The fee that we expect to pay is the sum of:
+        // - Revenue/reputation diff -> 1000000 - 700000 = 300000
+        // - Sum of risk of each htlc
+        // - Fee buffer -> 1000
+        let expected_fee = 300_000 + expected_htlc_risk + 1000;
```

Comment:
Perhaps add tests cases for:
- We have enough reputation and don't need to pay
- We have reputation that covers part of the `expected_htlc_risk` but not all 

Should be easy enough to turn this into a multi-case test.